README.md

```diff
@@ -8,4 +8,31 @@ Tap for Zendesk
 
 ---
 
+## Authentication
```

Comment:
This may be obvious to someone coming to this tap as a `zendesk` API user but I think it would be helpful to maintainers who are coming to this tap with little-to-no `zendesk` expertise to document why one might use one or the other and what the default one is etc.

Reply:
I'll improve this.